code/game/machinery/porta_turret/portable_turret.dm

```diff
@@ -921,3 +921,167 @@
 				on = 0
 				spawn(100)
 					on = 1
+
+/////// MANNED TURRET ////////
+
+/obj/machinery/manned_turret
+	name = "machine gun turret"
+	desc = "While the trigger is held down, this gun will redistribute recoil to allow its user to easily shift targets."
+	icon = 'icons/obj/turrets.dmi'
+	icon_state = "machinegun"
+	can_buckle = TRUE
+	density = TRUE
+	max_integrity = 100
+	obj_integrity = 100
+	buckle_lying = 0
+	layer = 4.2
+	var/view_range = 10
+	var/cooldown = 0
+	var/projectile_type = /obj/item/projectile/bullet/weakbullet3
+	var/rate_of_fire = 1
+	var/number_of_shots = 40
+	var/cooldown_duration = 90
+	var/atom/target
+	var/target_turf
+	var/warned = FALSE
+
+//BUCKLE HOOKS
+
+/obj/machinery/manned_turret/unbuckle_mob(mob/living/buckled_mob,force = 0)
+	playsound(src,'sound/mecha/mechmove01.ogg', 50, 1)
+	for(var/obj/item/I in buckled_mob.held_items)
+		if(istype(I, /obj/item/gun_control))
+			qdel(I)
+	if(istype(buckled_mob))
+		buckled_mob.pixel_x = 0
+		buckled_mob.pixel_y = 0
+		if(buckled_mob.client)
+			buckled_mob.client.change_view(world.view)
+	anchored = FALSE
+	. = ..()
+
+/obj/machinery/manned_turret/user_buckle_mob(mob/living/M, mob/living/carbon/user)
+	if(user.incapacitated() || !istype(user))
+		return
+	M.forceMove(get_turf(src))
+	..()
+	for(var/V in M.held_items)
+		var/obj/item/I = V
+		if(istype(I))
+			if(M.dropItemToGround(I))
+				var/obj/item/gun_control/TC = new /obj/item/gun_control(src)
+				M.put_in_hands(TC)
+		else	//Entries in the list should only ever be items or null, so if it's not an item, we can assume it's an empty hand
+			var/obj/item/gun_control/TC = new /obj/item/gun_control(src)
+			M.put_in_hands(TC)
+	M.pixel_y = 14
+	layer = 4.1
+	setDir(SOUTH)
+	playsound(src,'sound/mecha/mechmove01.ogg', 50, 1)
+	anchored = TRUE
+	if(user.client)
+		user.client.change_view(view_range)
+
+/obj/item/gun_control
+	name = "turret controls"
+	icon = 'icons/obj/weapons.dmi'
+	icon_state = "offhand"
+	w_class = WEIGHT_CLASS_HUGE
+	flags = ABSTRACT | NODROP
+	resistance_flags = FIRE_PROOF | UNACIDABLE | ACID_PROOF | NOBLUDGEON
+	var/obj/machinery/manned_turret/turret
+
+/obj/item/gun_control/New(var/obj/machinery/manned_turret/MT)

```

Comment:
been doing this as a precaution since I've seen it all over, thanks for clarifying